stack/framework/components/scheduler/scheduler.c

```diff
@@ -208,7 +208,7 @@ __LINK_C uint8_t get_task_id(task_t task)
 
 __LINK_C error_t sched_register_task(task_t task)
 {
-  assert(NG(num_registered_tasks) <= NUM_TASKS);
+  assert(NG(num_registered_tasks) < NUM_TASKS);
```

Comment:
This also gets checked on other lines in this file. Could you change those too?

Reply:
I think the other checks in `get_task_id()` and `check_structs_are_valid()` are actually correct as-is. The subtle difference is that the other checks are done in code paths that don't result in incrementing `NG(num_registered_tasks)`.

The check `NG(num_registered_tasks) <= NUM_TASKS` itself is valid, the problem specific to `sched_register_task` is that it performs this check before incrementing.

Reply:
You are right!